backend/models/accountPermissions.js

```diff
@@ -0,0 +1,102 @@
+/**
+ *  Copyright (C) 2019 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+"use strict";
+const responseCodes = require("../response_codes.js");
+const C = require("../constants");
+const { intersection } = require("lodash");
+const Project = require("./project");
+
+const updatePermissions = async function(teamspace, updatedPermissions) {
+	const User = require("./user");
+	await User.update(teamspace.user, {"customData.permissions": updatedPermissions});
+	return updatedPermissions;
+};
+
+const checkValidUpdate = async (teamspace, teamMember, permissions) => {
+	const User = require("./user");
+	const userToCheck = await User.findByUserName(teamMember);
+	if(!userToCheck) {
+		throw (responseCodes.USER_NOT_FOUND);
+	}
+
+	if(!User.isMemberOfTeamspace(userToCheck, teamspace.user)) {
```

Comment:
I don't think we need to do L32 anymore, 37 should just take in a username?

It's only coded this way because it was a mongoose method i believe

The `isMemberOfTeamspace` method should do L32-37 within the function

Reply:
isMemberOfTeamspace takes the user object, it checks for the roles from that user.

Reply:
that's what i mean, the only reason why this code needs to get the user object is to pass it in for isMemberOfTeamspace to use. It should instead take the teamspace as a string and does it itself